test/ZkMinterModTriggerV1.t.sol

```diff
@@ -1,343 +1,350 @@
-    }
+  function testFuzz_RevertIf_TriggerAfterContractIsClosed(address _caller) public {
+    vm.prank(admin);
+    minterTrigger.close();
+
+    vm.prank(_caller);
+    vm.expectRevert(ZkMinterV1.ZkMinter__ContractClosed.selector);
+    minterTrigger.trigger();
   }
 
-  function test_RevertWhen_ExceedsCap() public {
-    // Deploy a new caller with call data exceeding the cap
-    uint256 withdrawAmount = 600 ether; // Exceeds cap of 500 ether
-    address destination = address(0x456);
-    bytes32 leaf = keccak256(abi.encode(destination, withdrawAmount));
-    bytes32 merkleRoot = leaf;
-    bytes32 ipfsHash = keccak256("ipfs data");
-
-    address[] memory targetAddresses = new address[](2);
-    targetAddresses[0] = address(token);
-    targetAddresses[1] = address(target);
-
-    bytes[] memory functionSignatures = new bytes[](2);
-    functionSignatures[0] = abi.encodeWithSignature("approve(address,uint256)");
-    functionSignatures[1] = abi.encodeWithSignature("addMerkleTree(bytes32,bytes32,address,uint256)");
-
-    bytes[] memory callDatas = new bytes[](2);
-    callDatas[0] = abi.encode(address(target), uint256(600 ether)); // Approve 600 ether
-    callDatas[1] = abi.encode(merkleRoot, ipfsHash, address(token), uint256(600 ether)); // Try to setup tree with 600
-      // ether
-
-    ZkMinterModTriggerV1 badCaller =
-      new ZkMinterModTriggerV1(cappedMinterAdmin, targetAddresses, functionSignatures, callDatas);
-
-    // Configure minter and roles
-    vm.prank(cappedMinterAdmin);
-    badCaller.setMinter(address(cappedMinter));
-    vm.prank(cappedMinterAdmin);
-    cappedMinter.grantRole(MINTER_ROLE, address(badCaller));
-
-    // Should revert due to exceeding cap
-    vm.prank(user);
+  function testFuzz_RevertIf_NotMinter(address _nonMinter) public {
+    vm.assume(_nonMinter != caller && _nonMinter != admin);
+
+    vm.prank(_nonMinter);
+    vm.expectRevert(_formatAccessControlError(_nonMinter, MINTER_ROLE));
+    minterTrigger.trigger();
+  }
+
+  function test_RevertIf_FunctionCallFails() public {
+    address[] memory _targets = new address[](1);
+    _targets[0] = address(mockTarget);
+
+    bytes[] memory _calldatas = new bytes[](1);
+    _calldatas[0] = abi.encodeWithSelector(mockTarget.revertFunction.selector);
+
+    uint256[] memory _values = new uint256[](1);
+    _values[0] = 100 ether;
+
+    ZkMinterModTriggerV1 failTrigger = new ZkMinterModTriggerV1(mintable, admin, _targets, _calldatas, _values);
+
+    vm.prank(admin);
+    failTrigger.grantRole(MINTER_ROLE, caller);
+
     vm.expectRevert(
-      abi.encodeWithSignature("ZkCappedMinterV2__CapExceeded(address,uint256)", address(badCaller), 600 ether)
+      abi.encodeWithSelector(
+        ZkMinterModTriggerV1.ZkMinterModTriggerV1__TriggerCallFailed.selector, 0, address(mockTarget)
+      )
     );
-    badCaller.mint(600 ether);
+    vm.prank(caller);
+    failTrigger.trigger();
+  }
+}
+
+contract IntegrationTest is ZkMinterModTriggerV1Test {
```

Comment:
Integration tests should be moved to ZkMinterModTriggerV1.integration.t.sol. We have a good example of this with the eligibility module

Reply:
https://github.com/zksync-association/zkminters/pull/33/commits/66f0e144cdb546dcbef6972507cde547a33e75e3